Linter tests may be flaky

I just installed vscode-go and the required go toolchain on a clean kubuntu install. After running the tests I get the following assert:

```go
      AssertionError: too many errors [{"file":"/home/razael/go/src/test/testfixture/linterTest/linter_1.go","line":1,"msg":"don't use MixedCaps in package name; linterTest should be lintertest (golint)","severity":"warning"},{"file":"/home/razael/go/src/test/testfixture/linterTest/linter_2.go","line":1,"msg":"don't use MixedCaps in package name; linterTest should be lintertest (golint)","severity":"warning"},{"file":"/home/razael/go/src/test/testfixture/linterTest/linter_1.go","line":8,"msg":"error return value not checked (a declared but not used) (errcheck)","severity":"warning"},{"file":"/home/razael/go/src/test/testfixture/linterTest/linter_2.go","line":5,"msg":"error return value not checked (missing return) (errcheck)","severity":"warning"},{"file":"/home/razael/go/src/test/testfixture/linterTest/linter_1.go","line":5,"msg":"exported function ExportedFunc should have comment or be unexported (golint)","severity":"warning"}]
      + expected - actual
      -5
      +3
      
      at util_1.getGoVersion.then.goLint_1.goLint.then.diagnostics (DEV/GITHUB/vscode-go/test/go.test.ts:872:12)
      at <anonymous>
```

My guess is that my clean install of gometalinter brought some new linter rules that trigger the assert

We have some options on how to deal with this that I'm thinking of:
- Make it more lazy (don't assert length, or only match result file)
- Maintain the current expected errors but don't fail on unexpected ones

--- a/test/go.test.ts
+++ b/test/go.test.ts
@@ -268,15 +268,19 @@ It returns the number of bytes written and any write error encountered.
 			}
 			return check(vscode.Uri.file(path.join(fixturePath, 'errorsTest', 'errors.go')), config).then(diagnostics => {
 				let sortedDiagnostics = diagnostics.sort((a, b) => a.line - b.line);
-				assert.equal(sortedDiagnostics.length, expected.length, `too many errors ${JSON.stringify(sortedDiagnostics)}`);
+				assert.equal(sortedDiagnostics.length > 0, true, `Failed to get linter results`);
+				let matchCount = 0;
 				for (let i in expected) {
-					if (expected[i].line) {
-						assert(sortedDiagnostics[i]);
-						assert.equal(sortedDiagnostics[i].line, expected[i].line);
-					};
-					assert.equal(sortedDiagnostics[i].severity, expected[i].severity);
-					assert.equal(sortedDiagnostics[i].msg, expected[i].msg);
+					for (let j in sortedDiagnostics) {
+						if (expected[i].line
+						&& (expected[i].line === sortedDiagnostics[j].line)
+						&& (expected[i].severity === sortedDiagnostics[j].severity)
+						&& (expected[i].msg === sortedDiagnostics[j].msg)) {
+							matchCount++;
+						}
+					}
 				}
+				assert.equal(matchCount >= expected.length, true, `Failed to match expected errors`);
 			});
 		}).then(() => done(), done);
 	});
@@ -390,13 +394,18 @@ It returns the number of bytes written and any write error encountered.
 					return 0;
 				});
 
-				assert.equal(sortedDiagnostics.length, expected.length, `too many errors ${JSON.stringify(sortedDiagnostics)}`);
-
+				assert.equal(sortedDiagnostics.length > 0, true, `Failed to get linter results`);
+				let matchCount = 0;
 				for (let i in expected) {
-					assert.equal(sortedDiagnostics[i].line, expected[i].line, `Failed to match expected error #${i}: ${JSON.stringify(sortedDiagnostics)}`);
-					assert.equal(sortedDiagnostics[i].severity, expected[i].severity, `Failed to match expected error #${i}: ${JSON.stringify(sortedDiagnostics)}`);
-					assert.equal(sortedDiagnostics[i].msg, expected[i].msg, `Failed to match expected error #${i}: ${JSON.stringify(sortedDiagnostics)}`);
+					for (let j in sortedDiagnostics) {
+						if ((expected[i].line === sortedDiagnostics[j].line)
+						&& (expected[i].severity === sortedDiagnostics[j].severity)
+						&& (expected[i].msg === sortedDiagnostics[j].msg)) {
+							matchCount++;
+						}
+					}
 				}
+				assert.equal(matchCount >= expected.length, true, `Failed to match expected errors`);
 
 				return Promise.resolve();
 			});

